src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseParticipationService.java

```diff
@@ -339,12 +339,17 @@ public void unlockStudentRepository(ProgrammingExercise programmingExercise, Pro
     public void stashChangesInStudentRepositoryAfterDueDateHasPassed(ProgrammingExercise programmingExercise, ProgrammingExerciseStudentParticipation participation) {
         if (participation.getInitializationState().hasCompletedState(InitializationState.REPO_CONFIGURED)) {
             try {
-                // Note: exam exercise do not have a due date, this method should only be invoked directly after the due date so now check is needed here
-                Repository repo = gitService.getOrCheckoutRepository(participation);
-                gitService.stashChanges(repo);
+                if (programmingExercise.getDueDate().isBefore(ZonedDateTime.now())) {
+                    Repository repo = gitService.getOrCheckoutRepository(participation);
+                    gitService.stashChanges(repo);
+                }
+                else {
+                    log.warn("Cannot stash student repository for participation " + participation.getId() + " because the due date has not passed yet!");
+                }
+
             }
             catch (InterruptedException | GitAPIException e) {
-                log.error("Stashing student repository for participation " + participation.getId() + " in exercise '" + programmingExercise.getTitle()
+                log.error("Stashing student repository for participation" + participation.getId() + " in exercise '" + programmingExercise.getTitle()
```

Comment:
Is that change intentional? That will cramp `participation` and the id together.

Reply:
No somehow this is reverts a change that was done previously